webextension/background.js

```diff
@@ -38,37 +38,41 @@ class StudyLifeCycleHandler {
    * call `this.enableFeature` to actually do the feature/experience/ui.
    */
   constructor() {
+    // IMPORTANT:  Listen for onEndStudy first.
     browser.study.onEndStudy.addListener(this.handleStudyEnding);
     browser.study.onReady.addListener(this.enableFeature);
   }
 
   /**
-   * do some cleanup / 'feature reset'
+   * Cleanup
    *
    * (If you have privileged code, you might need to clean
    *  that up as well.
    * See:  https://firefox-source-docs.mozilla.org/toolkit/components/extensions/webextensions/lifecycle.html
+   *
+   * @returns {undefined}
    */
-  async cleanup() {
-    await browser.storage.local.clear();
+  cleanup() {
+    // do whatever work your addon needs to clean up
+    browser.cloudstorage.cleanUpPrefs();
   }
 
   /**
+   *
+   * side effects
    * - set up expiration alarms
    * - make feature/experience/ui with the particular variation for this user.
+   *
+   * @param {object} studyInfo browser.study.studyInfo object
+   *
+   * @returns {undefined}
    */
-  async enableFeature(studyInfo) {
-    if (studyInfo.timeUntilExpire) {
+  enableFeature(studyInfo) {
+    /* if (studyInfo.timeUntilExpire) {
       browser.alarms.create(studyInfo.timeUntilExpire, () =>
```

Comment:
@gregglind I have commented browser.alarms call as it's throwing undefined error. Is that needed?